### PR TITLE
Feature/interactive player 2.0.0

### PR DESCRIPTION
**Pull request by aravind-raveendran**
## Description

1. Use string identifiers for view's and use it to track active consumers of a specific video track
2. Reorganise files and folders
3. Removed DolbyIORTSUIKit
4. Refactor PIP implementation
5. Refactor PIP to reuse the same Video view for both display in app and in PIP (SDK limits a maximum of one renderer attached to the Video track in a given time)

### CODE REVIEW

**Review by djova-dolby — APPROVED** [review]

**Comment by aravind-raveendran** [comment]
I have noted an issue where the List view tile freezes the video, this is because of the below check. I will fix and test it when I am back.
```
        guard videoViewController.isPiPView != viewModel.isPiPView, videoViewController.renderer != viewModel.renderer else {
            return
        }
```

**Comment by aravind-raveendran** [comment]
@Yousif-CS - Looks like the flicker issue will still happen at least till we have the limit of one MCSampleBufferView for a CMSampleBuffer based renderer? Basically the flicker happens the more I create a new instance of MCSampleBufferView (in this case this is unavoidable as we reuse the same renderer for multiple views). See [commit](https://github.com/dolbyio-samples/rts-app-ios-viewer/pull/199/commits/3c531dfc2269f8170646b721da51cbdb545f410f).

**Comment by aravind-raveendran** [comment]
> @Yousif-CS - Looks like the flicker issue will still happen at least till we have the limit of one MCSampleBufferView for a CMSampleBuffer based renderer? Basically the flicker happens the more I create a new instance of MCSampleBufferView (in this case this is unavoidable as we reuse the same renderer for multiple views). See [commit](https://github.com/dolbyio-samples/rts-app-ios-viewer/pull/199/commits/3c531dfc2269f8170646b721da51cbdb545f410f).

Fixed.

**Comment by aravind-raveendran** [comment]
> I have noted an issue where the List view tile freezes the video, this is because of the below check. I will fix and test it when I am back.
> 
> ```
>         guard videoViewController.isPiPView != viewModel.isPiPView, videoViewController.renderer != viewModel.renderer else {
>             return
>         }
> ```

Fixed

File: interactive-player/Interactive Viewer/Managers/Persistence/CoreDataManager.swift [hunk 1]
```diff
@@ -27,7 +27,6 @@ final class CoreDataManager {
         }
         container.loadPersistentStores(completionHandler: { _, error in
             if let error = error {
-//                fatalError("Failed loading persistent stores with error: \(error.localizedDescription)")
                 print("$$$ \("Failed loading persistent stores with error: \(error.localizedDescription)")")
             }
```
Comment (sheiladoherty-dolby) [inline interactive-player/Interactive Viewer/Managers/Persistence/CoreDataManager.swift:31]: Remove comment. We should refrain from using `fatalError` on Apple apps. `fatalError` automatically causes the app to crash, which is never the desired behavior.

**Comment by sheiladoherty-dolby** [comment]
[question] How is PiP triggered? I turned it on in the settings & then backgrounded the app during playback & I got no PiP. I also did not see a button to trigger it manually? Also, when PiP is disabled & the app is backgrounded, there is still audio. Is this expected behavior?

**Comment by aravind-raveendran** [comment]
> [question] How is PiP triggered? I turned it on in the settings & then backgrounded the app during playback & I got no PiP. I also did not see a button to trigger it manually? Also, when PiP is disabled & the app is backgrounded, there is still audio. Is this expected behavior?

Did you test PIP in a device ? There is no button to trigger PIP its expected to trigger only on backgrounding. Playing Audio is expected in this app, we do not disable the tracks when there is no PIP

**Comment by sheiladoherty-dolby** [comment]
> > [question] How is PiP triggered? I turned it on in the settings & then backgrounded the app during playback & I got no PiP. I also did not see a button to trigger it manually? Also, when PiP is disabled & the app is backgrounded, there is still audio. Is this expected behavior?
> 
> Did you test PIP in a device ? There is no button to trigger PIP its expected to trigger only on backgrounding. Playing Audio is expected in this app, we do not disable the tracks when there is no PIP

Yes, I tested it on 2 different iPhones. Actual devices, not simulators

File: interactive-player/Interactive Viewer/Managers/Persistence/CoreDataManager.swift (hunk 1 above)
Comment (aravind-raveendran) [reply interactive-player/Interactive Viewer/Managers/Persistence/CoreDataManager.swift:31]: Isn't that already a line deletion ?

**Comment by aravind-raveendran** [comment]
Thanks for testing PIP in devices @sheiladoherty-dolby - I will look into the issue and open a followup PR for the fix.

**Pull request merged**